File: lib/limelight/builtin/players/button.rb
#- Copyright 2008 8th Light, Inc.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

module Limelight
  module Builtin
    module Players

      module Button
        class << self

          def extended(prop)
            button = Limelight::UI::Model::Inputs::ButtonPanel.new
            prop.panel.add(button)
            prop.button = button.button
          end
        end

        attr_accessor :button

      end

    end
  end
end

File: src/limelight/builtin/players/Button.java
//- Copyright © 2008-2011 8th Light, Inc. All Rights Reserved.
//- Limelight and all included source files are distributed under terms of the MIT License.

package limelight.builtin.players;

import limelight.ui.events.panel.PanelEvent;
import limelight.ui.model.PropPanel;
import limelight.ui.model.inputs.ButtonPanel;

public class Button
{
  private PropPanel propPanel;
  private ButtonPanel buttonPanel;

  public void install(PanelEvent event)
  {
    buttonPanel = new ButtonPanel();
    propPanel = (PropPanel)event.getRecipient();
    propPanel.add(buttonPanel);
    propPanel.getStagehands().put("button", this);
  }

  public PropPanel getPropPanel()
  {
    return propPanel;
  }

  public ButtonPanel getButtonPanel()
  {
    return buttonPanel;
  }
}
